pkg/util/config/config.go

```diff
@@ -45,5 +114,166 @@ func (c *GenericConfig) DeepCopy() *GenericConfig {
 
 func (c *GenericConfig) DeepCopyInto(out *GenericConfig) {
 	*out = *c
-	out.Config = c.DeepCopyJsonObject().Config
+	out.mp = c.DeepCopyJsonObject().mp
+}
+
+func (c *GenericConfig) Set(key string, value interface{}) {
+	set(c.mp, key, value)
+}
+
+func (c *GenericConfig) Get(key string) (value *Value) {
+	if c == nil {
+		return nil
+	}
+
+	v := get(c.mp, key)
+	if v == nil {
+		return nil
+	}
+
+	return &Value{inner: v}
+}
+
+type Value struct {
+	inner interface{}
+}
+
+func (v *Value) Interface() interface{} {
+	return v.inner
+}
+
+func (v *Value) MustString() string {
+	value, err := v.AsString()
+	if err != nil {
+		panic(err)
+	}
+	return value
+}
+
+func (v *Value) AsString() (string, error) {
+	s, ok := (v.inner).(string)
+	if !ok {
+		return "", errors.Errorf("type is %v not string", reflect.TypeOf(v.inner))
+	}
+	return s, nil
+}
+
+func (v *Value) MustInt() int64 {
+	value, err := v.AsInt()
+	if err != nil {
+		panic(err)
+	}
+	return value
+}
+
+func (v *Value) AsInt() (int64, error) {
+	switch value := v.inner.(type) {
+	case int:
+		return int64(value), nil
+	case int8:
+		return int64(value), nil
+	case int16:
+		return int64(value), nil
+	case int32:
+		return int64(value), nil
+	case int64:
+		return int64(value), nil
+	// return error if overflow ?
+	case uint:
+		return int64(value), nil
+	case uint8:
+		return int64(value), nil
+	case uint16:
+		return int64(value), nil
+	case uint32:
+		return int64(value), nil
+	case uint64:
+		return int64(value), nil
+
+	default:
+		return 0, errors.Errorf("type is %v not integer", reflect.TypeOf(v.inner))
+	}
+}
+
+func (v *Value) MustFloat() float64 {
+	value, err := v.AsFloat()
+	if err != nil {
+		panic(err)
```

Comment:
Is it reasonable to panic here?
This will cause the operator process panic, right?

Reply:
no code except test code will call this directly now.
as the name `Must..` you should call this only if it should be the expected type or it's a program error.

Reply:
ok